### PR TITLE
gsoc: Add apply-single-action

### PR DESCRIPTION
**Pull request by Nosferatul**

### CODE REVIEW

**Comment by Nosferatul** [comment]
@Adrianzatreanu @yukiisbored please check this http://imgur.com/C8k08US.
So how does this work. If the user uses `-res` or `--apply-patch` flag, first he can choose what action do you want to apply for all results. The choice will be applied for all results

File: coalib/output/ConsoleInteraction.py [hunk 1]
```diff
@@ -152,14 +153,26 @@ def acquire_actions_and_apply(console_printer,
...
         # User can always choose no action which is guaranteed to succeed
-        if not ask_for_action_and_apply(console_printer,
-                                        section,
-                                        metadata_list,
-                                        action_dict,
-                                        failed_actions,
-                                        result,
-                                        file_diff_dict,
-                                        file_dict):
+        if apply_single is not False:
```
Comment (Adrianzatreanu) [inline coalib/output/ConsoleInteraction.py]: So basically: "if apply_single"..?

Comment (Nosferatul) [inline coalib/output/ConsoleInteraction.py]: right, my bad

File: coalib/coala_main.py [hunk 2]
```diff
@@ -20,6 +20,21 @@ def do_nothing(*args):
     return True
 
 
+ALL_ACTIONS_POSSIBLE = ['Do (N)othing', '(A)pply patch', 'Add (I)gnore comment',
```
Comment (yukiisbored) [inline coalib/coala_main.py]: I assume this means it's impossible or impractical to collect action names with the current library :/

Comment (Nosferatul) [inline coalib/coala_main.py]: It's possible if we import all of the actions and just use for example `ShowPatchAction().get_metadata().desc` I'll need to do this for all actions. I think it's impractical but maybe i'm wrong.

Comment (yukiisbored) [inline coalib/coala_main.py]: Well, you can create a method that does it and stores it into a variable and when it retriggers again we can just give them the variable.

Comment (Nosferatul) [inline coalib/coala_main.py]: ok

File: coalib/parsing/DefaultArgParser.py [hunk 3]
```diff
@@ -231,6 +231,10 @@ def default_arg_parser(formatter_class=None):
         help="don't create .orig backup files before patching")
 
     misc_group.add_argument(
+        '-res', '--apply-patch', const=True, action='store_const',
```
Comment (yukiisbored) [inline coalib/parsing/DefaultArgParser.py]: Euhm, doesn't this conflict with `-a`, `--apply-patches`? I think it's confusing, I suggest `--single-action`, `-A`

**Comment by yukiisbored** [comment]
ack https://github.com/coala/coala/pull/4462/commits/54da68fa1cdbfe051ae3b31ad637bda794b778b2

**Comment by Adrianzatreanu** [comment]
Rebase it, reack it and merge it yourself @Nosferatul , you got my ack and yuki's

**Comment by Nosferatul** [comment]
reack 65feb30

**Comment by Nosferatul** [comment]
@rultor merge

**Pull request merged**